### PR TITLE
Add CTR Guide to EDU documentation

### PR DESCRIPTION
**Pull request by cppwfs**
resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/3104

### CODE REVIEW

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 1]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
 
 # Create and launch a Composed Task using Data Flow
 
-In this section we introduce composed tasks, show how to register task applications with Data Flow, create a composed task DSL, and launch the composed task to Cloud Foundry, Kubernetes and your local machine.
+In our previous sections we used two applications:
 
-## Development
+- billsetuptask: The billsetuptask setups up the database for the billrun application.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Maybe wrap in `billsetuptask` and `billrun`.

Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Maybe wrap in **billsetuptask**

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 2]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+- billrun: The billrun reads usage information from a file and generates a billing statement record for each entry from a usage file.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Maybe wrap in **billrun**

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 3]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
 
-Explaine what we are doing to do development wise.
+Previously we executed the billsetuptask app and then the billrun app in sequence manually.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: `billsetuptask` app and then the `billrun` app

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 4]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+This certainly works for the small case where we want to run 2 apps back to back where the first app runs in a few seconds.
+But what about the cases where there are 5, 10, 20 or even more apps that need to be run in a sequence and the runtime of some the apps maybe hours.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: maybe --> may be

Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: hours.  --> hours?

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 5]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+- billrun: The billrun reads usage information from a file and generates a billing statement record for each entry from a usage file.
 
-Explaine what we are doing to do development wise.
+Previously we executed the billsetuptask app and then the billrun app in sequence manually.
+This certainly works for the small case where we want to run 2 apps back to back where the first app runs in a few seconds.
+But what about the cases where there are 5, 10, 20 or even more apps that need to be run in a sequence and the runtime of some the apps maybe hours.  
+Or cases where one of the apps returns a non zero exit code. What do we do in this case?
+Spring Cloud Data Flow provides a solution for this issue through the use of Composed Tasks.  
+A composed task is a directed graph where each node of the graph is a task application.
+To use the case above a graph would look something like this:
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: above a graph --> above, a graph

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 6]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+![Bill Run Composed Task](images/SCDF-composed-task-simple.png)
 
-### Application Registration
+In this section we show how to create a composed task using Spring Cloud Data Flow's UI, and then launch the composed task to Cloud Foundry, Kubernetes and your local machine.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Should it be capitalized? : composed task --> Composed Task

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 7]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+### Register the Billing Applications
 
-## Deployment
+For this guide, we will need for you to have registered the [billsetuptask](%currentPath%/batch-developer-guides/batch/data-flow-simple-task) and the [billrun](%currentPath%/batch-developer-guides/batch/data-flow-spring-batch) applications in Spring Cloud Data Flow.
 
-Deploy to local, Cloud Foundry and Kubernetes
+### Register the Composed Task Runner
 
-### Local
+If you haven't already done so import the [Spring Cloud Task App Starters](https://cloud.spring.io/spring-cloud-task-app-starters/), which will give us the `composed-task-runner` application.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Add comma?? : If you haven't already done so import --> If you haven't already done so, import

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 8]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+To do this, select `Apps` from the left navigation bar, then select `Add Applications(s)`.
+When the `Add Application(s)` page appears select **Bulk import application coordinates from an HTTP URI location**.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Add comma? page appears select --> page appears, select

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 9]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
 
-### Kubernetes
+Select `Tasks` from the left navigation bar, then select `Create task(s)`.
+![Create Compose Task](images/SCDF-create-ctr.png)
 
-Where all the cool kids play.
+This displays a graphical editor that we can use to compose tasks.
+The initial canvas contains `START` and `END` nodes. To the left of the canvas, we see the available task applications, including `billsetuptask` and `billrun` apps.
+Drag the `billsetuptask` app to the canvas and connect the START node to the `billsetuptask`. Now drag that `billrun` application to the canvas. Connect `billsetuptask` to the `billrun` and the `billrun` to the END node to complete the composed task definition.
+In this case, the composed task definition consists of a two task apps (billsetuptask and billrun).
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: consists of a two task apps --> consists of two task apps

Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: `billsetuptask` and `billrun`

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 10]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+If there are app defined configuration properties, we would set them here.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: app defined --> app-defined

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 11]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+### Verify the results
+
+When the execution is complete, the status for each of the task definitions will turn to a satisfying green color and show `Complete.`
+Select the `Executions` tab to view a summary of executions for this task.
+
+![Task executions](images/SCDF-composed-executions.png)
+
+### Deleting a composed task
+
+As shown before a composed task creates a definition for each app that is in the graph.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Add comma: As shown before, a composed task

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 12]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+To delete all the definitions that were created as a part of the composed task we will delete the composed task runner definition.
+To do that we will select the checkbox next to the ct-statement definition, then press the actions button at the upper left hand corner and select the `Destroy Tasks` option.
```
Comment (ghillert) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: `Actions` button

**Comment by cppwfs** [comment]
Comments addressed and rebased.

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 13]
```diff
@@ -6,36 +6,111 @@ description: 'Create and launch a Composed Task using Data Flow'
 
 # Create and launch a Composed Task using Data Flow
 
-In this section we introduce composed tasks, show how to register task applications with Data Flow, create a composed task DSL, and launch the composed task to Cloud Foundry, Kubernetes and your local machine.
+In our previous sections we used two applications:
 
-## Development
+- `billsetuptask`: The `billsetuptask` setups up the database for the `billrun` application.
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Perhaps:

> `billsetuptask`: The `billsetuptask` configures the ..

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 14]
```diff
@@ -6,36 +6,111 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+<!--END_TIP-->
+
+### Launch the Task
+
+![View the tasks](images/SCDF-composed-task-list.png)
+
+We see that three task definitions were created when we created the ct-statement.
+
+- ct-statement: This definition represents the composed task runner that will execute the tasks in the composed task graph.
+- ct-statement-billsetuptask: This definition represents the billsetuptask app that will be executed.
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Maybe with backticks: `billsetuptask`

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 15]
```diff
@@ -6,36 +6,111 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+- ct-statement-billrun: This definition represents the billrun app that will be executed.
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: It'd be nice with backticks: `billrun`

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 16]
```diff
@@ -6,36 +6,111 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+
+Now we will launch the ct-statement by pressing the `play` button (that's the middle icon that looks like an arrow head pointing right).
+This will take you to a form where you can add command line arguments and deployment parameters, but we don't need any for this composed task.
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: typo: form -> from

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 17]
```diff
@@ -6,36 +6,111 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+
+<!--TIP-->
+
+**TIP:** If it is required that a Composed Task needs to be executed multiple times then the `increment-instance-enabled` property needs to be set to true for each launch.
+
+<!--END_TIP-->
+
+### Verify the results
+
+When the execution is complete, the status for each of the task definitions will turn to a satisfying green color and show `Complete.`
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: Nicely written!

File: content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md [hunk 18]
```diff
@@ -6,36 +6,105 @@ description: 'Create and launch a Composed Task using Data Flow'
...
+In this case, we will use the same name as the task application which is `ct-statement`.
+
+![Confirm create task](images/SCDF-composed-task-confirmation.png)
+
+Press `Create the task`.
+This will display the main `Tasks` view.
+
+<!--TIP-->
+
+**TIP:** To see all the things that can be done with the Composed Task Graph take a look at the Composed Task [section](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_composed_tasks_dsl) of the reference documentation.
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply content/documentation/pages/4-batch-developer-guides/2-batch/5-data-flow-composed-task.md]: I'm following the plan outlined in the syntax guide.

**Comment by ghillert** [comment]
LGTM - polished 2 images during merge. Merged with commit 374a16b371d407698dd707065f843e94cf68669f

**Pull request closed without merging**